### PR TITLE
cassandra 3.11.3 -> 3.11.4

### PR DESCRIPTION
**Pull request by contrun**
###### Motivation for this change
update cassandra

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by nlewo** [comment]
@GrahamcOfBorg test cassandra

**Comment by aanderse** [comment]
Looks like the cassandra test may have been removed from `nixos/tests/all-tests.nix` at some point? @contrun do you mind adding the cassandra test back to this file so we can run tests here?

**Comment by contrun** [comment]
@aanderse Looks like the only file in the nixos/tests directory contains cassandra is nixos/tests/cassandra.nix. I tried to search cassandra with `git grep cassandra $(git rev-list --all -- nixos/tests/) -- nixos/tests/ | awk -F: '{print $2}' | sort | uniq`.

**Comment by aanderse** [comment]
@contrun pardon my poor choice of wording. The actual test remains, but the test is not actually executed by the bot because a reference to the test was removed (or never existed?) in all-tests.nix.

Can you please create an additional commit adding the test back into all-tests.nix?

**Comment by contrun** [comment]
@aanderse Added that. But I am not familiar with the testing process. Not sure if it will work.

**Comment by aanderse** [comment]
I've been running the test from the master branch locally for ~10 minutes now and no success.

@cransom as the listed maintainer of the cassandra package do you have any information which could help us?

**Comment by aanderse** [comment]
@GrahamcOfBorg test cassandra

**Comment by aanderse** [comment]
@JohnAZoidberg as someone interested in cassandra do you have any insight into why the test is not succeeding?

**Comment by JohnAZoidberg** [comment]
Seems like you can't have a node as its own seed when it's listening on an interface instead of an IP.
At least the way it's done in the current test doesn't work and I can't figure out any other way.

I'll push a fix to my PR from yesterday that improves the config file: #59179

**Comment by aanderse** [comment]
@contrun Any chance I can get you to leave a review and approval on https://github.com/NixOS/nixpkgs/pull/59179 if everything looks good to you? Bonus points if you test out the changes in your own configuration and mention results. Once https://github.com/NixOS/nixpkgs/pull/59179 is merged it seems like you could rebase this PR on master and we could *quickly* and confidently merge this. Sorry for the hold up so far.

**Comment by JohnAZoidberg** [comment]
And if you want you can upgrade the other supported versions we have to their new point release that came out on the same day: https://github.com/apache/cassandra/releases

**Comment by aanderse** [comment]
This PR can be merged now as there is no blocker anymore, right? 100% relying on you @JohnAZoidberg to tell me whether to push the merge button.

@contrun Are you able to rebase on master and possibly update other versions as requested by @JohnAZoidberg?

**Comment by JohnAZoidberg** [comment]
@GrahamcOfBorg test cassandra

**Review by JohnAZoidberg — APPROVED** [review]
The tests take too long for OfBorg. They run fine on my machine.

**Comment by aanderse** [comment]
@GrahamcOfBorg build cassandra

**Comment by aanderse** [comment]
Merging solely based on the good word of @JohnAZoidberg.

**Pull request merged**
